client/service/src/task_manager/mod.rs

```diff
@@ -127,7 +124,19 @@ impl SpawnTaskHandle {
 			}
 		};
 
-		let join_handle = self.executor.spawn(Box::pin(future.in_current_span()), task_type);
+		let future: BoxFuture<()> = if let Some(telemetry_span) = self.telemetry_span.clone() {
+			// This will preserve the telemetry span and by extension all of its 
+			// parents as they are linked together upon creation.
+			// Used by [`sc_telemetry::layer::TelemetryLayer`].
+			Box::pin(future.instrument(telemetry_span.span()))
+		} else {
+			// This will preserve the current span (if any) and by extension all 
+			// of its parents as they are linked together upon creation.
+			// Used by [`sc_tracing::logging::prefix_logs_with`].
+			Box::pin(future.in_current_span())
+		};
```

Comment:
I still don't see how this works when I have some custom span (that I have created) and the telemetry is active. It would not be handled here. Can we not do the following?

```suggestion
		{
			let _ span = self.telemetry_span.as_ref().map(|s| s.enter());
			let join_handle = self.executor.spawn(Box::pin(future.in_current_span()), task_type);
		}
```

Reply:
I think your concerns are right :+1:  

I triple-checked the solution. It's hard to test because I saw the issue only in cumulus (probably because of how the things are started) and I confirm your solution works and it's simpler and I don't have to add a comment on the code.

Thanks!!